test/entry.ts

```diff
@@ -345,7 +357,11 @@ describe('Entry', () => {
     it('should convert a string timestamp', () => {
       entry.metadata.timestamp = new Date();
       const json = entry.toStructuredJSON();
-      assert(withinExpectedTimeBoundaries(new Date(json.timestamp!)));
+      assert(
+        withinExpectedTimeBoundaries(
+          new Date(nanosecondsToDate(json.timestamp!))
+        )
+      );
```

Comment:
It might be good to add another test to ensure that `json.timestamp` is now in the expected dictionary format.  `nanosecondsToDate()` would probably throw some errors if it wasn't, but it still might be good to be safe

Reply:
Thats the test which exactly testing the format - see `nanosecondsToDate` function.

Reply:
`nanosecondsToDate` isn't actually an explicit test itself though. Nothing is explicitly testing the `toNanosAndSecondsObj` function. We could be testing to make sure that `.nanos` and `.seconds` are floats instead of ints or strings, for example

Reply:
I must admit it seems like overkill... I believe format tests for structures could be part of proto testing, but I think failure in `nanosecondsToDate` is good enough as indicator for wrong format.